crowbar_framework/spec/lib/crowbar/lock/shared_non_blocking_spec.rb

```diff
@@ -0,0 +1,92 @@
+require "spec_helper"
+require "timeout"
+require "rspec/expectations"
+
+RSpec::Matchers.define :be_locked do
+  match do |actual|
+    actual.locked? && File.exists?(actual.path)
+  end
+end
+
+RSpec::Matchers.define :be_unlocked do
+  match do |actual|
+    ! actual.locked? && ! File.exists?(actual.path)
+  end
+end
+
+describe Crowbar::Lock::SharedNonBlocking do
+  let(:klass) { Crowbar::Lock::SharedNonBlocking }
+  let(:local_executor) { lambda { |node, cmd| system(cmd) } }
+  let(:defaults) {
+    {
+      executor: local_executor,
+      node: 'dummy', # local_executor ignores this
+      owner: 'rspec',
+      reason: 'testing',
+    }
+  }
+  let(:lock) { new_lock }
+
+  def new_lock(options={})
+    options[:path] ||= Dir::Tmpname.create(['test_shared_nb-', '.lock']) { }
```

Comment:
Style/SpaceInsideBlockBraces: Space inside empty braces detected.<br>Style/StringLiterals: Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.